### PR TITLE
add group permissions documentation

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #207
# What's Changed

- Add group permissions documentation to readthedocs

localhost_8001_administration_permissions_

# TODO
- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/administration/permissions.md [hunk 1]
```diff
@@ -46,3 +46,41 @@ Site.objects.filter(
...
+Permissions can be applied directly to users or to groups of users.
+
+### Permissions Assigned to Individual Users
+
+Permissions can be related directly to users from the Admin UI or the API:
+
+| -                                                           | Admin UI | API |
+| ----------------------------------------------------------- | -------- | --- |
+| Superusers                                                  | Yes      | Yes |
+| Users with permission to add or change `users | permission` | No       | Yes |
```
Comment (glennmatthews) [inline nautobot/docs/administration/permissions.md:61]: Did we discuss whether this discrepancy between UI and API should be considered a bug?

File: nautobot/docs/administration/permissions.md [hunk 2]
```diff
@@ -46,3 +46,41 @@ Site.objects.filter(
...
+
+!!! info
+
+    User permission relationships can be managed in the Admin UI by modifying the user or the permission.
+
+!!! warning
+
+    Granting a user permission to add or change `users | permission` gives the user the ability to modify their own permissions. This permission should be restricted to trusted accounts and should be considered the same as giving a user full access.
```
Comment (glennmatthews) [inline nautobot/docs/administration/permissions.md:69]: Similar warning needed about `auth|group` permissions?

File: nautobot/docs/administration/permissions.md [hunk 3]
```diff
@@ -46,3 +46,41 @@ Site.objects.filter(
...
+| Superusers                                                  | Yes      | Yes |
+| Users with permission to add or change `auth | group`       | No       | Yes |
+
+!!! info
+
+    Group permission relationships can be managed in the Admin UI by modifying the group or the permission.
+
+### Group Assignment
+
+Users can be added to groups through the Admin UI by superusers or automatically assigned to externally authenticated users through the [`EXTERNAL_AUTH_DEFAULT_GROUPS`](../../configuration/optional-settings/#external_auth_default_groups) setting.
```
Comment (glennmatthews) [inline nautobot/docs/administration/permissions.md:86]: Possibly also reference `AUTH_LDAP_FIND_GROUP_PERMS` (nautobot/docs/configuration/authentication/ldap.md) and `EXTERNAL_AUTH_DEFAULT_PERMISSIONS` settings as well?

File: nautobot/docs/administration/permissions.md (hunk 1 above)
Comment (gsnider2195) [reply nautobot/docs/administration/permissions.md:61]: I'll open an issue so we can discuss

Comment (gsnider2195) [reply nautobot/docs/administration/permissions.md:61]: Opened issue #1839

**Comment by gsnider2195** [comment]
Updated permissions doc

localhost_8001_administration_permissions_ (2)

**Comment by bryanculver** [comment]
I think this is sufficient for now but I would like to see a more user guide style doc eventually that helps new users walk through a setup:

1. Create Permission
    1. Give name
    2. Select actions
    3. Select object types
    4. Optional: apply constraints
    5. Save
2. Create Group
    1. Give name
    2. Select permission name(s)
    3. Save
3. Create/Edit User
    1. Assign user to group, OR
    2. Assign user to permission

**Review by bryanculver — APPROVED** [review]

**Pull request merged**